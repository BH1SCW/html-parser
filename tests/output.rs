use html_parser::{HtmlParser, Result};
use indoc::indoc;
use insta::assert_json_snapshot;

#[test]
fn it_can_output_json() -> Result<()> {
    assert!(HtmlParser::parse("<div/>")?.to_json().is_ok());
    Ok(())
}

#[test]
fn it_can_output_json_pretty() -> Result<()> {
    assert!(HtmlParser::parse("<div/>")?.to_json_pretty().is_ok());
    Ok(())
}

#[test]
fn it_can_output_complex_html_as_json() -> Result<()> {
    let markup = indoc!(
        "<html lang=\"sv\">
        <head>
            <title>Här kan man va</title>
        </head>
            <body>
                <h1>Tjena världen!</h1>
                <p>Tänkte bara informera om att Sverige är bättre än Finland i ishockey.</p>
            </body>
        </html>"
    );
    let ast = HtmlParser::parse(markup)?;
    assert_json_snapshot!(ast);
    Ok(())
}
