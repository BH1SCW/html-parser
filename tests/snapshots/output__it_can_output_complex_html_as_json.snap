---
source: tests/output.rs
expression: ast
---
{
  "treeType": "document",
  "nodes": [
    {
      "element": {
        "id": null,
        "name": "html",
        "variant": "normal",
        "attributes": {
          "lang": "sv"
        },
        "classes": [],
        "nodes": [
          {
            "element": {
              "id": null,
              "name": "head",
              "variant": "normal",
              "attributes": {},
              "classes": [],
              "nodes": [
                {
                  "element": {
                    "id": null,
                    "name": "title",
                    "variant": "normal",
                    "attributes": {},
                    "classes": [],
                    "nodes": [
                      {
                        "text": "Här kan man va"
                      }
                    ]
                  }
                }
              ]
            }
          },
          {
            "element": {
              "id": null,
              "name": "body",
              "variant": "normal",
              "attributes": {},
              "classes": [],
              "nodes": [
                {
                  "element": {
                    "id": null,
                    "name": "h1",
                    "variant": "normal",
                    "attributes": {},
                    "classes": [],
                    "nodes": [
                      {
                        "text": "Tjena världen!"
                      }
                    ]
                  }
                },
                {
                  "element": {
                    "id": null,
                    "name": "p",
                    "variant": "normal",
                    "attributes": {},
                    "classes": [],
                    "nodes": [
                      {
                        "text": "Tänkte bara informera om att Sverige är bättre än Finland i ishockey."
                      }
                    ]
                  }
                }
              ]
            }
          }
        ]
      }
    }
  ]
}
